File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        mp = defaultdict(int)
        for n in nums:
            mp[n]+=1
        heap = list((-value, key) for key, value in mp.items())
        heapq.heapify(heap)
        result = []
        for i in range(k):
            result.append(heapq.heappop(heap)[1])
        return result

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int l = 0, r = nums.size()-1;

        int ret = nums[0];
        while (l <= r) {
            if (nums[l] < nums[r]) {
                ret = min(ret, nums[l]);
                break;
            }

            int m = l + (r-l)/2;
            ret = min(ret, nums[m]);

            if (nums[l] <= nums[m]) {
                // [2 3 4 5 6 1]
                l = m+1;
            } else if (nums[l] > nums[m]) {
                // [6 1 2 3 4 5]
                r = m-1;
            }
        }

        return ret;
    }    
};
